tests/vmi_multus_test.go

```diff
@@ -35,102 +36,243 @@ import (
-			detachedVMI.Namespace = "default"
 			detachedVMI.Spec.Domain.Devices.Interfaces = []v1.Interface{{Name: "default", InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}},
 				{Name: "ptp", InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}}}
 			detachedVMI.Spec.Networks = []v1.Network{{Name: "default", NetworkSource: v1.NetworkSource{Pod: &v1.PodNetwork{}}},
 				{Name: "ptp", NetworkSource: v1.NetworkSource{Multus: &v1.MultusNetwork{NetworkName: "ptp-conf"}}}}
 
-			_, err = virtClient.VirtualMachineInstance("default").Create(detachedVMI)
+			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(detachedVMI)
 			Expect(err).ToNot(HaveOccurred())
-			tests.WaitUntilVMIReadyWithNamespace("default", detachedVMI, tests.LoggedInCirrosExpecter)
+			tests.WaitUntilVMIReady(detachedVMI, tests.LoggedInCirrosExpecter)
 
-			cmdCheck := fmt.Sprintf("ping %s -c 1 -w 5\n", "10.1.1.1")
+			cmdCheck := "sudo /sbin/cirros-dhcpc up eth1 > /dev/null\n"
 			err = tests.CheckForTextExpecter(detachedVMI, []expect.Batcher{
 				&expect.BSnd{S: "\n"},
 				&expect.BExp{R: "\\$ "},
 				&expect.BSnd{S: cmdCheck},
 				&expect.BExp{R: "\\$ "},
-				&expect.BSnd{S: "echo $?\n"},
-				&expect.BExp{R: "0"},
-			}, 180)
+				&expect.BSnd{S: "ip addr show eth1 | grep 10.1.1 | wc -l"},
+				&expect.BExp{R: "1"},
+			}, 15)
 			Expect(err).ToNot(HaveOccurred())
 
 			By("checking virtual machine instance as two interfaces")
-			cmdCheck = fmt.Sprintf("ip link show %s\n", "eth0")
-			err = tests.CheckForTextExpecter(detachedVMI, []expect.Batcher{
-				&expect.BSnd{S: "\n"},
-				&expect.BExp{R: "\\$ "},
-				&expect.BSnd{S: cmdCheck},
-				&expect.BExp{R: "\\$ "},
-				&expect.BSnd{S: "echo $?\n"},
-				&expect.BExp{R: "0"},
-			}, 180)
+			checkInterface(detachedVMI, "eth0", "\\$ ")
+			checkInterface(detachedVMI, "eth1", "\\$ ")
+
+			pingVirtualMachine(detachedVMI, "10.1.1.1", "\\$ ")
+		})
+	})
+
+	Context("VirtualMachineInstance with ovs-cni plugin interface", func() {
+		nodeAffinity := &k8sv1.Affinity{
+			NodeAffinity: &k8sv1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &k8sv1.NodeSelector{
+					NodeSelectorTerms: []k8sv1.NodeSelectorTerm{
+						{
+							MatchExpressions: []k8sv1.NodeSelectorRequirement{
+								{Key: "kubernetes.io/hostname", Operator: k8sv1.NodeSelectorOpIn, Values: []string{nodes.Items[0].Name}},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		AfterEach(func() {
+			deleteVMIs(virtClient, []*v1.VirtualMachineInstance{vmiOne, vmiTwo})
+		})
+
+		It("should create two virtual machines with one interface", func() {
+			By("checking virtual machine instance can ping the secondary virtual machine instance using ovs-cni plugin")
+			vmiOne = tests.NewRandomVMIWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskAlpine), "#!/bin/bash\n")
```

Comment:
aren't they both the same? cannot create them using an inline function?

Reply:
Done